src/components/Onboarding/Screens/Apps/TokensSettingsNative.js

```diff
@@ -0,0 +1,417 @@
+  if (members.some(([account, stake]) => !(stake > 0))) {
+    return 'Every account has to have a positive balance.'
+  }
+  if (!validateDuplicateAddresses(members)) {
+    return 'One of your members is using the same address than another member. Please ensure every member address is unique.'
+  }
+  if (!tokenName.trim()) {
+    return 'Please add a token name.'
+  }
+  if (!tokenSymbol) {
+    return 'Please add a token symbol.'
+  }
+  return null
+}
+
+function TokensSettingsNative() {
+  const theme = useTheme()
+  const fieldsLayout = useFieldsLayout()
+
+  const {
+    config,
+    onBack,
+    onNext,
+    steps,
+    step,
+    onConfigChange,
+  } = useOnboardingState()
+
+  const [formError, setFormError] = useState()
+
+  const [tokenName, setTokenName] = useState(config.tokens.name)
+  const [tokenSymbol, setTokenSymbol] = useState(config.tokens.symbol)
+
+  const [members, setMembers] = useState(
+    config.tokens.holders.length === 0 ? [['', 0]] : config.tokens.holders
+  )
+
+  const handleTokenNameChange = useCallback(event => {
+    setFormError(null)
+    setTokenName(event.target.value)
+  }, [])
+
+  const handleTokenSymbolChange = useCallback(event => {
+    setFormError(null)
+    setTokenSymbol(event.target.value.trim().toUpperCase())
+  }, [])
+
+  const membersRef = useRef()
+  const [focusLastMemberNext, setFocusLastMemberNext] = useState(false)
+
+  useEffect(() => {
+    if (!focusLastMemberNext || !membersRef.current) {
+      return
+    }
+
+    setFocusLastMemberNext(false)
+
+    // This could be managed in individual MemberField components, but using
+    // the container with a .member class makes it simpler to manage, since we
+    // want to focus in three cases:
+    //   - A new field is being added.
+    //   - A field is being removed.
+    //   - The first field is being emptied.
+    //
+    const elts = membersRef.current.querySelectorAll('.member')
+    if (elts.length > 0) {
+      elts[elts.length - 1].querySelector('input').focus()
+    }
+  }, [focusLastMemberNext])
+
+  const focusLastMember = useCallback(() => {
+    setFocusLastMemberNext(true)
+  }, [])
+
+  const addMember = useCallback(() => {
+    setFormError(null)
+    setMembers(members => [...members, ['', 0]])
+    focusLastMember()
+  }, [focusLastMember])
+
+  const removeMember = useCallback(
+    index => {
+      setFormError(null)
+      setMembers(members =>
+        members.length < 2
+          ? // When the remove button of the last field
+            // gets clicked, we only empty the field.
+            [['', 0]]
+          : members.filter((_, i) => i !== index)
+      )
+      focusLastMember()
+    },
+    [focusLastMember]
+  )
+
+  const updateMember = useCallback((index, updatedAccount, updatedStake) => {
+    setFormError(null)
+    setMembers(members =>
+      members.map((member, i) =>
+        i === index ? [updatedAccount, updatedStake] : member
+      )
+    )
+  }, [])
+
+  const handleNext = useCallback(
```

Comment:
One thing that I notice with a final review is that when we do `onBack` we are not storing the config changes if they were any. I don't think this is particularly bad behavior. Just wanted to bring it up to have your oppinions. 

cc @fabriziovigevani

Reply:
Regardless I think this PR is adding a lot of value. We can create a new one with that specific change on every `onBack` across screens if we decided to.

Reply:
That's right. I think we'll need to create a PR to fix it on every `onBack`. It is not such a rare practice, when using a installation wizard, moving backwards and performing changes. It would be nice to have support for it.